dotnet/src/SemanticKernel.Abstractions/Search/ITextSearchResultMapper.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Interface for mapping between a text search data model, and a <see cref="TextSearchResult" /> instance.
+/// </summary>
+[Experimental("SKEXP0001")]
+public interface ITextSearchResultMapper
+{
+    /// <summary>
+    /// Map from the text search result to a a <see cref="TextSearchResult" /> instance.
+    /// </summary>
+    /// <param name="result">The instance of the text search result to map.</param>
+    /// <returns>A <see cref="TextSearchResult" /> instance.</returns>
+    TextSearchResult MapFromResultToTextSearchResult(object result);
```

Comment:
What possible values can be passed as `object result` argument? XML doc says it's `instance of the text search result`, so I'm wondering if we should make it generic with some constraints? Another reason to use generics instead of `object` here is because if we pass value type parameter - `int`, `bool`, `struct`, in case of `object` it will perform type boxing, which may impact the performance, while in case of generics no boxing will be needed.

Reply:
The `object result` will be something like a `BingWebPage` or a Google `Result` or Azure AI Search `SearchDocument`. 

We're not using generic in ITextSearch so introducing it for the mappers feels awkward. I have updated the API documentation to better match the terminology used elsewhere.